src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/LdapAccessPolicyDAO.scala

```diff
@@ -247,7 +247,60 @@ class LdapAccessPolicyDAO(
         ldapSearchStream(resourceDn(resource), SearchScope.SUB, Filter.createEqualityFilter("objectclass", ObjectClass.policy))(unmarshalAccessPolicy)
       ))
 
-  override def listAccessPoliciesForUser(resource: FullyQualifiedResourceId, user: WorkbenchUserId): IO[Set[AccessPolicy]] =
+
+  private def isUserMemberOfPolicy(policy: AccessPolicy, userId: WorkbenchUserId): IO[Boolean] = {
+    if (policy.members.contains(userId)) {
+      IO.pure(true)
+    } else {
+      // KCIBUL -- what
+      policy.members.collect{case x:WorkbenchGroupName => x}.toList.existsM(isUserMemberOfGroup(_, userId))
```

Comment:
re: your "what" comment ... this code looks fine to me!

Reply:
ah that was pre my discovery of collect :D